### PR TITLE
Add GDPR utility

### PR DESCRIPTION
**Pull request by clarascu**
## Description

Feature

#### What does this PR do?
Adds annotations that mark GDPR related fields, and endpoint to retrieve data.

#### Breaking changes
None

#### Code review focus
If there's an edge case, that is not covered properly.

### Checks

- [x] Tested with Oracle/PostgreSQL
- [x] Documentation added
- [x] Tests added
- [x] E2e tests created
- [ ] Successfully ran e2e tests before merge

closes GH-77

### CODE REVIEW

File: src/main/java/at/ac/tuwien/damap/repo/GdprRepo.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    private GdprResult getGdprEntityDataByUniversityId(HqlQuery query, String universityId) {
+
+        GdprResult gdprResult = new GdprResult();
+        gdprResult.setEntity(query.getEntityName());
+
+        @SuppressWarnings("unchecked")
+        List<Map<String, Object>> result = entityManager.createQuery(query.getHql(), (Class<Map<String, Object>>) (Class<?>) Map.class)
+                .setParameter("id", universityId).getResultList();
+        result.forEach(map -> map.values().removeIf(Objects::isNull));
```
Comment (ZenoLC) [inline src/main/java/at/ac/tuwien/damap/repo/GdprRepo.java:34]: please add a comment detailing the behavior of this removal, as on first glance i was worried we might be removing key information here (like personID, which is relevant result itself)
```suggestion
        //check if any of the principal output/result entities (e.g. Access, Consent, Contributor) are empty, and remove them if so.
```

**Review by ZenoLC — APPROVED** [review]
looks nicely generalized
